qiita_db/metadata_template.py

```diff
@@ -1590,6 +1590,25 @@ def create(cls, md_template, raw_data, study, data_type,
             conn_handler.execute(
                 "DELETE FROM qiita.prep_template where "
                 "{0} = %s".format(cls._id_column), (prep_id,))
+
+            # Check if sample IDs present here but not in sample template
+            sql = ("SELECT sample_id from qiita.required_sample_info WHERE "
+                   "study_id = %s")
+            # Get list of study sample IDs, prep template study IDs,
+            # and their intersection
+            study_samples = set(s[0] for s in conn_handler.execute_fetchall(
```

Comment:
The cast is so you can do the set operations (difference_update, etc) on it, not to filter to distinct samples.